x/programs/program/function.go

```diff
@@ -40,37 +39,18 @@ func (f *Func) Call(context Context, params ...uint32) ([]int64, error) {
 	if err != nil {
 		return nil, err
 	}
-	contextPtr, err := writeToMem(context, mem)
+	contextPtr, err := context.WriteToMem(mem)
 	if err != nil {
 		return nil, err
 	}
 
-	result, err := f.inner.Call(f.inst.GetStore(), append([]interface{}{int32(contextPtr)}, callParams...)...)
-	if err != nil {
+	if _, err := f.inner.Call(f.inst.GetStore(), append([]interface{}{int32(contextPtr)}, callParams...)...); err != nil {
 		return nil, HandleTrapError(err)
 	}
-	switch v := result.(type) {
-	case int32:
-		value := int64(result.(int32))
-		return []int64{value}, nil
-	case int64:
-		value := result.(int64)
-		return []int64{value}, nil
-	case nil:
-		// the function had no return values
-		return nil, nil
-	default:
-		return nil, fmt.Errorf("invalid result type: %T", v)
-	}
-}
-
-func writeToMem(obj interface{}, memory *Memory) (uint32, error) {
-	bytes, err := borsh.Serialize(obj)
-	if err != nil {
-		return 0, err
-	}
 
-	return AllocateBytes(bytes, memory)
+	result := context.Result()
+	context.ClearResult()
```

Comment:
Is that used to "clear" memory as we don't have garbage collection since this is ran in WebAssembly ? I thought that there was no way to deallocate memory in WASM, could that potentially be used as a "memory bomb attack" ?